pkg/query/endpointset.go

```diff
@@ -345,28 +354,125 @@ func (e *EndpointSet) Update(ctx context.Context) {
 				"address", addr, "extLset", extLset, "duplicates", fmt.Sprintf("%v", stats[component.Sidecar][extLset]+stats[component.Rule][extLset]+1))
 		}
 		stats[er.ComponentType()][extLset]++
+		if _, ok := currentEndpoints[er.addr]; !ok {
+			level.Info(e.logger).Log("msg", fmt.Sprintf("adding new %v with %+v", er.ComponentType(), er.apisPresent()), "address", addr, "extLset", extLset)
```

Comment:
Why adding this here and not when we actually added it?

Reply:
I think I just took this over from the previous implementation. Will remove.

Reply:
What I've also noticed is that these logs can be very verbose when the labelset is present. So we might want to log the IP and supported APIs, at least with info level?